---
title: "R Notebook for RNAseq Diaphragm Analysis in Schweizer et al. (submitted)"
output:
  html_document:
    toc: yes
    df_print: paged
  html_notebook: default
  
---

This is an [R Markdown](http://rmarkdown.rstudio.com) Notebook for the analysis of RNAseq data generated from Ivy et al. 2021 and published in Schweizer et al. (submitted).

Start by making sure all variables are cleared, then load libraries for analysis. 

```{r load_packages}
message = FALSE # hide package loading messages
rm(list=ls())
library(lme4)
library(lmerTest)
library(edgeR)
library(DESeq2)
library(WGCNA)
library(Hmisc)
library(reshape)
library(gprofiler2)
library(ggbiplot)
library(plotrix)
library(UpSetR)
library(VennDiagram)
library(RColorBrewer)
library(emmeans)
library(qwraps2)
library(viridis)
```


We start by importing our technical sequencing metrics data for our RNAseq samples. We will then run principal components analysis and generate a 95% data ellipse to identify technical outliers. The default stat_ellipse() is intended to compute a bivariate confidence interval assuming a Student-t distribution. There is more information on this process here https://stackoverflow.com/questions/16659749/getting-the-ellipse-function-to-match-the-dataellipse-function-in-r. 

We also run ANOVAs of sequencing metrics between lanes to test for batch effects. 

```{r read_in_metrics}
run_date <- format(Sys.Date(), "%d%b%Y")

metrics <- read.table("../data_files/rnaseq_qc_metrics_02Feb2022_noPH.txt",header=TRUE)
metrics$group_for_pca <- "samples"
metrics.pca <- prcomp(metrics[,c(5:7,9:10)], center = TRUE,scale. = TRUE) #don't include perc dups from fastqc or # reads 
summary(metrics.pca)
#str(metrics.pca)

# output to Console
ggbiplot(metrics.pca, ellipse=TRUE, labels=metrics$Sample_Name,groups=metrics$group_for_pca,circle=FALSE,colour="black",ellipse.prob = 0.95) + scale_colour_manual(values=c("black")) + ggtitle("PCA of RNASeq samples with 95% ellipse")

# make output figure file
pdf(file=paste("PCA_qc_metrics_oneGroup_95perc_noDups_noReads_",run_date,".pdf",sep=""),height=8,width=8)
ggbiplot(metrics.pca, ellipse=TRUE, labels=metrics$Sample_Name,groups=metrics$group_for_pca,circle=FALSE,colour="black",ellipse.prob = 0.95) + scale_colour_manual(values=c("black")) + ggtitle("PCA of RNASeq samples with 95% ellipse")
dev.off()

# Run anova to check for batch effects

lm_qs <- lm(mean_quality_score ~ group2, data=metrics)
lm_q30 <- lm(perc_bases_above_q30 ~ group2, data=metrics)
lm_gc <- lm(perc_gc ~ group2, data=metrics)
lm_align <- lm(align_rate ~ group2, data=metrics)
lm_assign <- lm(assign_rate ~ group2, data=metrics)

anova(lm_qs)
anova(lm_q30)
anova(lm_gc)
anova(lm_align)
anova(lm_assign)

# remove 4 samples and calculate averages for metrics

metrics_mod <- metrics[metrics$Sample_Name!=c("AH1"),]
metrics_mod <- metrics_mod[metrics_mod$Sample_Name!=c("AH2"),]
metrics_mod <- metrics_mod[metrics_mod$Sample_Name!=c("N1"),]
metrics_mod <- metrics_mod[metrics_mod$Sample_Name!=c("N2"),]


set.seed(42)
  library(qwraps2)
# define the markup language we are working in.
# options(qwraps2_markup = "latex") is also supported.
options(qwraps2_markup = "markdown")

our_summary1 <-
  list("Mean Quality Score" =
       list("min"       = ~ min(mean_quality_score),
            "max"       = ~ max(mean_quality_score),
            "mean (sd)" = ~ qwraps2::mean_sd(mean_quality_score)),
       "Percent Bases Above Q30" =
       list("min"       = ~ min(perc_bases_above_q30),
            "median"    = ~ median(perc_bases_above_q30),
            "max"       = ~ max(perc_bases_above_q30),
            "mean (sd)" = ~ qwraps2::mean_sd(perc_bases_above_q30)),
       "GC (%)" =
       list("min"       = ~ min(perc_gc),
            "median"    = ~ median(perc_gc),
            "max"       = ~ max(perc_gc),
            "mean (sd)" = ~ qwraps2::mean_sd(perc_gc)),
       "Reads" =
       list("min"       = ~ min(reads),
            "median"    = ~ median(reads),
            "max"       = ~ max(reads),
            "mean (sd)" = ~ qwraps2::mean_sd(reads)),
       "Genome Alignment Rate (%)" =
       list("min"       = ~ min(align_rate),
            "max"       = ~ max(align_rate),
            "mean (sd)" = ~ qwraps2::mean_sd(align_rate)),
        "Assignment Rate (%)" =
       list("min"       = ~ min(assign_rate),
            "max"       = ~ max(assign_rate),
            "mean (sd)" = ~ qwraps2::mean_sd(assign_rate))
       )

### Overall
whole <- summary_table(metrics_mod, our_summary1)
whole

```
Based on the 95% PCA, we identify four outliers: AH1, AH2, N1, N2. We identify significant lane differences of quality score metrics and mapping rate, but no difference for GC content or featureCount aassignment rate. 

Import the raw counts generated by featureCounts. 

```{r read_in_raw__counts}
counts0 <- read.table("../data_files/pman_diaphragm_mp20_wMt_readcounts_noCommandLine_27Jan2022.txt",header=T) #load the counts.text file from featureCounts

counts <- counts0[, -c(2:6)] # remove the unncessary columns
rownames(counts) <- counts$Geneid #make the rownames the gene id's
counts$Geneid <- NULL #remove the gene id column, since you only want data columns in your matrix

```

We use EdgeR to create log transformed and normalized counts for WGCNA. We also remove our outlier samples and do some filtering for minimum average read counts. 

```{r filter_samples_and_low_reads}
id <-  read.table("../data_files/pman_diaphragm_metaData_06Sep2022.txt",header=T,sep="\t")

#STOP: make sure the column names in the counts dataset are ordered by sample ID
colnames(counts) == id$Code # should see all "TRUE"s 

dim(counts) # verify how many genes (first value) and samples (second value) we have data for

#remove 4 outlier samples identified from technical PCA
id <- id[id$Code!=c("AH1"),]
id <- id[id$Code!=c("AH2"),]
#id <- id[id$Code!=c("AH5"),]
id <- id[id$Code!=c("N1"),]
id <- id[id$Code!=c("N2"),]

#counts <- subset(counts, select=-c(AH1,AH2,AH5,N1,N2))
counts <- subset(counts, select=-c(AH1,AH2,N1,N2))
#colnames(counts) == id$Code # should see all "TRUE"s

# will not use PT samples in WGCNA or EdgeR
id <- id[id$Code!=c("PT1"),]
id <- id[id$Code!=c("PT2"),]
id <- id[id$Code!=c("PT3"),]
id <- id[id$Code!=c("PT4"),]
id <- id[id$Code!=c("PT5"),]
id <- id[id$Code!=c("PT6"),]
id <- id[id$Code!=c("PT7"),]

counts <- subset(counts, select=-c(PT1,PT2,PT3,PT4,PT5,PT6,PT7))


#filter data
counts$mean = rowMeans(counts) #rowMeans takes means of each row
keep_counts = subset(counts, mean >= 20) #filter out genes that have <20 reads on average across individuals
#dim(keep_counts)

keep_counts$mean = NULL #clean up dataset
dim(keep_counts)
```

Next, we set up our GLM model and use edgeR to normalize across libraries. 

```{r setup_GLM_normalize_libraries}

id$Code == colnames(keep_counts) # double check that the files are ordered in the same way, other than the "mean" column in the counts df
treatment <- id$Treatment #this is the first factor for the GLM model

table <- data.frame(treatment)
group <- factor(table$treatment)
cbind(table, group=group)
table$treatment = as.factor(table$treatment)
table$treatment <- relevel(table$treatment, ref="Control")#set Control as the reference level for population
design <- model.matrix(~0+treatment, data=table)
#colnames(design) <- c("N","AH","LH","PT","PNH") # change column names to be simpler

colnames(design) <- c("N","AH","LH","PNH") # change column names to be simpler

# do normalization 

y <- DGEList(counts=keep_counts, group=group) # make a DGE list
dim(y)
y <- calcNormFactors(y) # normalize
y <- estimateGLMCommonDisp(y, design, verbose=TRUE)
y <- estimateGLMTrendedDisp(y, design)
y <- estimateGLMTagwiseDisp(y, design)

# Generate MDS plot

pch <- c(0,1,2,16) #pch <- c(0,1,2,15,16)
colors<- (c("dark green","red","blue","black"))

plotMDS(y,col=colors[group],pch=pch[group],labels=colnames(y)) #exploration plots
legend("topleft",legend=levels(group),pch=pch,col=colors,ncol=2,cex=0.8)

pdf(file=paste("Rplot_MDS_n35_",run_date,".pdf",sep=""),width=8,height=8)
plotMDS(y,col=colors[group],pch=pch[group],labels=colnames(y)) #exploration plots
legend("topleft",legend=levels(group),pch=pch,col=colors,ncol=2,cex=0.8)
dev.off()


```

```{r begin_WGCNA_code_to_check_samples}

diaphragm.norm <- cpm(y,log=TRUE,prior.count=2,normalized.lib.sizes=TRUE) #cpm normalized and log transformed expression dataset
write.csv(diaphragm.norm, file=paste("wgcna_output_files/man.diaphragm.norm_droppedSamples_",run_date,".csv",sep=""))
diaphragm.norm1 <- diaphragm.norm
#transpose expression data for further analysis
Expr0 = as.data.frame(t(diaphragm.norm1));
rownames(Expr0) = colnames(diaphragm.norm1)
#check for genes and samples with too many missing values
gsg = goodSamplesGenes(Expr0, verbose = 0);
gsg$allOK
if (!gsg$allOK)
{
  # Optionally, print the gene and sample names that were removed:
  if (sum(!gsg$goodGenes)>0)
    printFlush(paste("Removing genes:", paste(names(Expr0)[!gsg$goodGenes], collapse = ", ")));
  if (sum(!gsg$goodSamples)>0)
    printFlush(paste("Removing samples:", paste(rownames(Expr0)[!gsg$goodSamples], collapse = ", ")));
  # Remove the offending genes and samples from the data:
  Expr0 = Expr0[gsg$goodSamples, gsg$goodGenes]
}
Expr = Expr0

#cluster the samples to check for outliers
sampleTree = hclust(dist(Expr0), method = "average");
# Plot the sample tree: Open a graphic output window of size 12 by 9 inches
# The user should change the dimensions if the window is too large or too small.
sizeGrWindow(12,9)

par(cex = 0.6);
par(mar = c(0,4,2,0))
plot(sampleTree, main = "Sample clustering to detect outliers", sub="", xlab="", cex.lab = 1.5,
     cex.axis = 1.5, cex.main = 2)

pdf(file = paste("wgcna_output_files/sampleClustering_n35_",run_date,".pdf",sep=""), width = 12, height = 9);
par(cex = 0.6);
par(mar = c(0,4,2,0))
plot(sampleTree, main = "Sample clustering to detect outliers", sub="", xlab="", cex.lab = 1.5,
     cex.axis = 1.5, cex.main = 2)
dev.off()

# Choose a set of soft-thresholding powers
powers = c(c(1:10), seq(from = 11, to=30, by=1))
# Call the network topology analysis function
sft = pickSoftThreshold(Expr, powerVector = powers, verbose = 0)
# Plot the results:
pdf(paste("wgcna_output_files/rv_beta_plot_droppedSamples_",run_date,".pdf",sep=""), h=7, w=7)
par(mfrow = c(1,2));
cex1 = 0.9;

# Scale-free topology fit index as a function of the soft-thresholding power
plot(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
     xlab="Soft Threshold (power)",ylab="Scale Free Topology Model Fit,signed R^2",type="n",
     main = paste("Scale independence"));
text(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
     labels=powers,cex=cex1,col="red");
# this line corresponds to using an R^2 cut-off of h
abline(h=0.8,col="red")
# Mean connectivity as a function of the soft-thresholding power
plot(sft$fitIndices[,1], sft$fitIndices[,5],
     xlab="Soft Threshold (power)",ylab="Mean Connectivity", type="n",
     main = paste("Mean connectivity"))
text(sft$fitIndices[,1], sft$fitIndices[,5], labels=powers, cex=cex1,col="red")
dev.off()


```

Construct a gene network, using a soft threshold power of 13, based on inflection point in figure. 

```{r construct_gene_network_WGCNA}

# construct gene network @ change the power! Previously used saveTOMs = TRUE and saveTOMFileBase="ExprTOM"
Net = blockwiseModules(Expr, power = 13, maxBlockSize = 16000, # set power according to threshold power plot
                       TOMType = "signed", networkType = "signed",
                       minModuleSize = 30,
                       reassignThreshold = 0, mergeCutHeight = 0.25,
                       numericLabels = TRUE, pamRespectsDendro = FALSE,
                       saveTOMs = TRUE,
                       saveTOMFileBase = "ExprTOM",
                       #loadTOMs = TRUE,
                       verbose = 0,
                       randomSeed=46693) #set seed so that results are fully reproducible
#table(Net$colors)
moduleLabels = Net$colors
moduleColors = labels2colors(Net$colors)
MEs = Net$MEs;
geneTree = Net$dendrograms[[1]];

table(moduleColors)

# circle plot with sizes proportional to number of genes in each module

plot_colors = sort(unique(moduleColors))
pdf(file=paste("wgcna_output_files/Rplot_moduleSizeColors_",run_date,".pdf",sep=""),h=9,w=9)
plot(1:30,type="n")
for (x in 1:length(table(moduleColors))) {
  if (table(moduleColors)[x]>1000){
    draw.circle(x*2,x*2,(table(moduleColors)[x])/2000,col=plot_colors[x])
    text(x*2,x*2+.25,labels=plot_colors[x],cex=0.8)
    text(x*2,x*2-.25,labels=table(moduleColors)[x],cex=0.8)
  }
  if (table(moduleColors)[x]<1000){
    draw.circle(x*2,x*2,(table(moduleColors)[x])/1000,col=plot_colors[x])
    text(x*2,x*2+.25,labels=plot_colors[x],cex=0.8)
    text(x*2,x*2-.25,labels=table(moduleColors)[x],cex=0.8)
  }}
dev.off()



```

Analysis of phenotypic correlations with module expression in WGCNA. 

```{r trait_correlation_analysis_WGCNA}
# load trait data
# 2 samples don't have phenotype data (LH4,PNH6), 
id <- id[id$Code!=c("LH4"),]
id <- id[id$Code!=c("PNH6"),]
traitData <- id

# Form a data frame analogous to expression data that will hold the phenotypic traits.
row_names_df_to_remove<-c("LH4","PNH6")
Expr <- Expr[!(row.names(Expr) %in% row_names_df_to_remove),]
Samples = rownames(Expr);
traitRows = match(Samples, traitData$Code)# make sure IDs are in the same order as Expr dataset
Traits0 = traitData[traitRows, -1]; # removes Code column
Traits0 <- na.omit(Traits0)
rownames(Traits0) <-  na.omit(traitData[traitRows, 1]);
Traits0$code<- rownames(Traits0)
Traits<- Traits0[,-c(1:7)]
#Traits = Traits0[, -c(1,3:4)]
#Traits$geno <- NULL
collectGarbage();
#match(rownames(Traits), rownames(Expr))# should be in order if datasets are matched

# Trait Module Associations
# Define numbers of genes and samples
nGenes = ncol(Expr);
nSamples = nrow(Expr)
# Recalculate MEs with color labels
MEs0 = moduleEigengenes(Expr, moduleColors)$eigengenes
MEs = orderMEs(MEs0) #the rownames of this dataset are equal to Expr
MEs1 <- MEs
#match(rownames(Traits), rownames(Expr))
moduleTraitCor = cor(MEs1, Traits, use = "p");
moduleTraitPvalue = corPvalueStudent(moduleTraitCor, nSamples);
corr.table <- as.data.frame(cbind(moduleTraitCor, moduleTraitPvalue))
colnames(corr.table) <- c("cor", "p")
corr.table$p.adjust <- p.adjust(corr.table$p, method="fdr")
corr.table <- corr.table[order(corr.table$p),]

# names (colors) of the modules
modNames = substring(names(MEs), 3)
geneModuleMembership = as.data.frame(cor(Expr, MEs, use = "p"));
MMPvalue = as.data.frame(corPvalueStudent(as.matrix(geneModuleMembership), nSamples));
names(geneModuleMembership) = paste("MM", modNames, sep="");
names(MMPvalue) = paste("p.MM", modNames, sep="");
# Create a dataset containing all gene-specific information
genes=names(Expr)
geneInfo0 = data.frame(Gene = genes,
                       moduleColor = moduleColors)
# Add module membership information in the chosen order
oldNames = names(geneInfo0)
geneInfo0 = data.frame(geneInfo0, geneModuleMembership,
                       MMPvalue);
names(geneInfo0) = c(oldNames, paste("MM.", modNames, sep=""),
                     paste("p.MM.", modNames, sep=""))

# Order the genes in the geneInfo variable first by module color, then by geneTraitSignificance
geneOrder = order(geneInfo0$moduleColor);
geneInfo = geneInfo0[geneOrder, ]

write.csv(geneInfo, file=paste("wgcna_output_files/man.diaphragm.norm_moduleExpression_genes_",run_date,".csv",sep=""))

#code to generate plot of module-trait relationships

pdf(paste("wgcna_output_files/module_trait_relationships_plot_",run_date,".pdf",sep=""), h=10, w=16)
#sizeGrWindow(10,6)
# Will display correlations and their p-values
textMatrix = paste(signif(moduleTraitCor, 2), "\n(",
                   signif(moduleTraitPvalue, 1), ")", sep = "");
dim(textMatrix) = dim(moduleTraitCor)
par(mar = c(12, 8.5, 3, 3));
# Display the correlation values within a heatmap plot
labeledHeatmap(Matrix = moduleTraitCor,
               xLabels = c("Diaphragm","VO2max","Arterial.O2","Heart.Rate","Breath.Freq","Tidal.Vol","Total.Vent","Air.Conv","Pul.Extract","Breath.Freq.8kPa","Tidal.Vol.8kPa","Total.Vent.8kPa","Breath.Freq.12kPa","Tidal.Vol.12kPa","Total.Vent.12kPa","Breath.Freq.21kPa","Tidal.Vol.21kPa","Total.Vent.21kPa","code"),
               yLabels = names(MEs),
               ySymbols = names(MEs),
               colorLabels = FALSE,
               colors = blueWhiteRed(50),
               textMatrix = textMatrix,
               setStdMargins = FALSE,
               cex.text = 1,
               zlim = c(-1,1),
               main = paste("Module-trait relationships"))
dev.off()

# and a version for right here
sizeGrWindow(6,10)
textMatrix = paste(signif(moduleTraitCor, 2), "\n(",
                   signif(moduleTraitPvalue, 1), ")", sep = "");
dim(textMatrix) = dim(moduleTraitCor)
par(mar = c(12, 8.5, 3, 3));
# Display the correlation values within a heatmap plot
labeledHeatmap(Matrix = moduleTraitCor,
               xLabels = c("Diaphragm","VO2max","Arterial.O2","Heart.Rate","Breath.Freq","Tidal.Vol","Total.Vent","Air.Conv","Pul.Extract","Breath.Freq.8kPa","Tidal.Vol.8kPa","Total.Vent.8kPa","Breath.Freq.12kPa","Tidal.Vol.12kPa","Total.Vent.12kPa","Breath.Freq.21kPa","Tidal.Vol.21kPa","Total.Vent.21kPa","code"),
               yLabels = names(MEs),
               ySymbols = names(MEs),
               colorLabels = FALSE,
               colors = blueWhiteRed(50),
               textMatrix = textMatrix,
               setStdMargins = FALSE,
               cex.text = 0.5,
               zlim = c(-1,1),
               main = paste("Module-trait relationships"))

# and a version of the correlations and p-values as a table to publish
colnames(moduleTraitCor) <- colnames(moduleTraitPvalue) <- c("Diaphragm","VO2max","Arterial.O2","Heart.Rate","Breath.Freq","Tidal.Vol","Total.Vent","Air.Conv","Pul.Extract","Breath.Freq.8kPa","Tidal.Vol.8kPa","Total.Vent.8kPa","Breath.Freq.12kPa","Tidal.Vol.12kPa","Total.Vent.12kPa","Breath.Freq.21kPa","Tidal.Vol.21kPa","Total.Vent.21kPa","code")
moduleTraitCor_mod <- merge(signif(moduleTraitCor[,1:18],2),signif(moduleTraitPvalue[,1:18],2),by='row.names',suffixes=c(".cor",".pval"))
write.table(moduleTraitCor_mod,file=paste("wgcna_output_files/module_trait_relationships_table_",run_date,".txt",sep=""),sep="\t",quote=FALSE,row.names=FALSE,col.names=TRUE)
```

We identified phenotypic traits with significant correlations with WGCNA modules. Next, we want to test for a significant effect of treatment on module expression using ANOVA tests. 

```{r run_ANOVAs_on_trait_correlations}

## code from Plachetzki et al 2014

## ANOVA to test effect of treatment 

ME = MEs
ME <- removeGreyME(MEs, greyMEName = paste(moduleColor.getMEprefix(), "grey", sep=""))

#head(ME)
ME_t <- ME
ME_t$Code <- row.names(ME_t)
ME_info <- merge(ME_t,id,by="Code")

mods <- colnames(ME)

# sig phenotype correlations at FDR <=0.05:  Turquoise/M2 (arterial O2), yellow/M4 (diaphragm), blue/M1 (arterial O2), brown/M3 (arterial 02) 
# no sig phenotype correlations: green

model_turq <- lmer(ME_info$MEturquoise ~ ME_info$Treatment + (1 | ME_info$Family))
model_yell <- lmer(ME_info$MEyellow ~ ME_info$Treatment + (1 | ME_info$Family))
model_green <- lmer(ME_info$MEgreen ~ ME_info$Treatment + (1 | ME_info$Family))
model_blue <- lmer(ME_info$MEblue ~ ME_info$Treatment + (1 | ME_info$Family))
model_brown <- lmer(ME_info$MEbrown ~ ME_info$Treatment+ (1 | ME_info$Family)) 

turq_anova <- anova(model_turq)
yell_anova <- anova(model_yell)
green_anova <- anova(model_green) # n.s.
blue_anova <- anova(model_blue) 
brown_anova <- anova(model_brown)

# put anova results in a table
anova_results <- data.frame(module=c("blue/M1","turquoise/M2","brown/M3","yellow/M4","green/M5"), DenDF=c(round(blue_anova$DenDF,3),round(turq_anova$DenDF,3),round(brown_anova$DenDF,3),round(yell_anova$DenDF,3),round(green_anova$DenDF,3)),Fval=c(round(blue_anova$`F value`,3),round(turq_anova$`F value`,3),round(brown_anova$`F value`,3),round(yell_anova$`F value`,3),round(green_anova$`F value`,3)),Pval=c(round(blue_anova$`Pr(>F)`,5),round(turq_anova$`Pr(>F)`,5),round(brown_anova$`Pr(>F)`,5),round(yell_anova$`Pr(>F)`,5),round(green_anova$`Pr(>F)`,5)))

write.table(anova_results,file = paste("wgcna_output_files/table_wgcna_anova_results_",run_date,".txt",sep=""), quote=FALSE,sep="\t",row.names=FALSE)

model_turq_emm <- emmeans(model_turq, pairwise ~ Treatment, adjust = "tukey") # M2
eff_size(model_turq_emm, sigma=sigma(model_turq),edf=27)

model_blue_emm <-emmeans(model_blue, pairwise ~ Treatment, adjust = "tukey") # M1
eff_size(model_blue_emm, sigma=sigma(model_blue),edf=27)

model_yell_emm <-emmeans(model_yell, pairwise ~ Treatment, adjust = "tukey") # M4
eff_size(model_yell_emm, sigma=sigma(model_yell),edf=27)

model_brown_emm <-emmeans(model_brown, pairwise ~ Treatment, adjust = "tukey") # M3
eff_size(model_brown_emm, sigma=sigma(model_brown),edf=27)

#match(rownames(Expr), id$Code)
setSamples = rownames(Expr)

```

We'll generate box plots of expression differences for modules. Gene expression of turquoise (M2), blue (M1), yellow (M4), and brown (M3) modules are significantly affected by treatment group. 

```{r box_plots_of_eigengene_expression}

# turquoise (M2), blue (M1), yellow (M4), and brown (M3) are significant with ANOVA


# Box plots

pdf(file=paste("wgcna_output_files/eigengene_expr_MEblue_",run_date,".pdf",sep=""),h=10, w=6)
ggplot(ME_info, aes(x=Treatment_for_plotting, y=MEblue)) +
  geom_boxplot() + geom_dotplot(binaxis='y', stackdir='center',
                                position=position_dodge(1), dotsize = 0.5)
dev.off()

pdf(file=paste("wgcna_output_files/eigengene_expr_MEturquoise_",run_date,".pdf",sep=""),h=10, w=6)
ggplot(ME_info, aes(x=Treatment_for_plotting, y=MEturquoise)) +
  geom_boxplot() + geom_dotplot(binaxis='y', stackdir='center',
                                position=position_dodge(1), dotsize = 0.5)
dev.off()

pdf(file=paste("wgcna_output_files/eigengene_expr_MEyellow_",run_date,".pdf",sep=""),h=10, w=6)
ggplot(ME_info, aes(x=Treatment_for_plotting, y=MEyellow)) +
  geom_boxplot() + geom_dotplot(binaxis='y', stackdir='center',
                                position=position_dodge(1), dotsize = 0.5)
dev.off()

pdf(file=paste("wgcna_output_files/eigengene_expr_MEbrown_",run_date,".pdf",sep=""),h=10, w=6)
ggplot(ME_info, aes(x=Treatment_for_plotting, y=MEbrown)) +
  geom_boxplot() + geom_dotplot(binaxis='y', stackdir='center',
                                position=position_dodge(1), dotsize = 0.5)
dev.off()

# output box plots of non-sig ANOVA but don't output to console (green)

pdf(file=paste("wgcna_output_files/eigengene_expr_MEgreen_",run_date,".pdf",sep=""),h=10, w=6)
ggplot(ME_info, aes(x=Treatment_for_plotting, y=MEgreen)) +
  geom_boxplot() + geom_dotplot(binaxis='y', stackdir='center',
                                position=position_dodge(1), dotsize = 0.5)
dev.off()

#pdf(file=paste("wgcna_output_files/eigengene_expr_MEyellow_",run_date,".pdf",sep=""),h=10, w=6)
#ggplot(ME_info, aes(x=Treatment, y=MEyellow)) +
#  geom_boxplot() + geom_dotplot(binaxis='y', stackdir='center',
#                                position=position_dodge(1), dotsize = 0.5)
#dev.off()

```
Output box plots of phenotype data by treatment. Diaphragm, arterial O2, tidal vol at 12kPa, total ventilation at 12kPa, tidal vol at 21kPa, and total ventiltion at 21kPa have significant correlations with WGCNA modules. 

```{r box_plots_of_phenotypes}

#box plots of phenotypes

pdf(file=paste("wgcna_output_files/phenotype_Diaphragm_",run_date,".pdf",sep=""),h=10, w=6)
ggplot(ME_info, aes(x=Treatment, y=Diaphragm_perc_body_mass)) +
  geom_boxplot() + geom_dotplot(binaxis='y', stackdir='center',
                                position=position_dodge(1), dotsize = 0.5)
dev.off()

pdf(file=paste("wgcna_output_files/phenotype_ArterialO2_",run_date,".pdf",sep=""),h=10, w=6)
ggplot(ME_info, aes(x=Treatment, y=Arteria_O2_perc)) +
  geom_boxplot() + geom_dotplot(binaxis='y', stackdir='center',
                                position=position_dodge(1), dotsize = 0.5)
dev.off()

pdf(file=paste("wgcna_output_files/phenotype_TidalVol_12kPa_",run_date,".pdf",sep=""),h=10, w=6)
ggplot(ME_info, aes(x=Treatment, y=Tidal.volume_ul_per_g_12kPa)) +
  geom_boxplot() + geom_dotplot(binaxis='y', stackdir='center',
                                position=position_dodge(1), dotsize = 0.5)
dev.off()

pdf(file=paste("wgcna_output_files/phenotype_TotalVent_12kPa_",run_date,".pdf",sep=""),h=10, w=6)
ggplot(ME_info, aes(x=Treatment, y=Total._ventilation_ml_per_g_per_min_12kPa)) +
  geom_boxplot() + geom_dotplot(binaxis='y', stackdir='center',
                                position=position_dodge(1), dotsize = 0.5)
dev.off()

pdf(file=paste("wgcna_output_files/phenotype_TidalVol_21kPa_",run_date,".pdf",sep=""),h=10, w=6)
ggplot(ME_info, aes(x=Treatment, y=Tidal.volume_ul_per_g_21kPa)) +
  geom_boxplot() + geom_dotplot(binaxis='y', stackdir='center',
                                position=position_dodge(1), dotsize = 0.5)
dev.off()

pdf(file=paste("wgcna_output_files/phenotype_TotalVent_21kPa_",run_date,".pdf",sep=""),h=10, w=6)
ggplot(ME_info, aes(x=Treatment, y=Total._ventilation_ml_per_g_per_min_21kPa)) +
  geom_boxplot() + geom_dotplot(binaxis='y', stackdir='center',
                                position=position_dodge(1), dotsize = 0.5)
dev.off()

#pdf(file=paste("wgcna_output_files/phenotype_HeartRate_",run_date,".pdf",sep=""),h=10, w=6)
#ggplot(ME_info, aes(x=Treatment, y=Hear_rate_bpm)) +
#  geom_boxplot() + geom_dotplot(binaxis='y', stackdir='center',
#                                position=position_dodge(1), dotsize = 0.5)
#dev.off()

#pdf(file=paste("wgcna_output_files/phenotype_BreathingFrequency_",run_date,".pdf",sep=""),h=10, w=6)
#ggplot(ME_info, aes(x=Treatment, y=Breathing_frequency_breath_per_min)) +
#  geom_boxplot() + geom_dotplot(binaxis='y', stackdir='center',
#                                position=position_dodge(1), dotsize = 0.5)
#dev.off()

```

Output scatterplots of module expression with phenotypes. 

```{r scatterplots_of_mod_w_phenotype}
pdf(file=paste("wgcna_output_files/phenotype_ArterialO2_MEblue_",run_date,".pdf",sep=""),h=4, w=3)
ggplot(ME_info, aes(x=MEblue, y=Arteria_O2_perc, shape=Treatment, color=Treatment)) +   geom_point(size=3) + geom_smooth(method = lm, aes(group=1))  + theme_bw() + scale_color_manual(values=c("#b2df8a","black", "#a6cee3","#1f78b4")) + theme(legend.position = "none",panel.grid = element_blank())
dev.off()

pdf(file=paste("wgcna_output_files/phenotype_ArterialO2_MEturquoise_",run_date,".pdf",sep=""),h=4, w=3)
ggplot(ME_info, aes(x=MEturquoise, y=Arteria_O2_perc, shape=Treatment, color=Treatment)) +   geom_point(size=3) + geom_smooth(method = lm, aes(group=1))  + theme_bw() + scale_color_manual(values=c("#b2df8a","black", "#a6cee3","#1f78b4")) + theme(legend.position = "none",panel.grid = element_blank())
dev.off()

pdf(file=paste("wgcna_output_files/phenotype_ArterialO2_MEbrown_",run_date,".pdf",sep=""),h=4, w=3)
ggplot(ME_info, aes(x=MEbrown, y=Arteria_O2_perc, shape=Treatment, color=Treatment)) +   geom_point(size=3) + geom_smooth(method = lm, aes(group=1))  + theme_bw() + scale_color_manual(values=c("#b2df8a","black", "#a6cee3","#1f78b4")) + theme(legend.position = "none",panel.grid = element_blank())
dev.off()

pdf(file=paste("wgcna_output_files/phenotype_Diaphragm_MEyellow_",run_date,".pdf",sep=""),h=4, w=3)
ggplot(ME_info, aes(x=MEyellow, y=Diaphragm_perc_body_mass, shape=Treatment, color=Treatment)) +   geom_point(size=3) + geom_smooth(method = lm, aes(group=1))  + theme_bw() + scale_color_manual(values=c("#b2df8a","black", "#a6cee3","#1f78b4")) + theme(legend.position = "none",panel.grid = element_blank())
dev.off()

pdf(file=paste("wgcna_output_files/phenotype_TidalVol_12kPa_MEyellow_",run_date,".pdf",sep=""),h=4, w=3)
ggplot(ME_info, aes(x=MEyellow, y=Tidal.volume_ul_per_g_12kPa, shape=Treatment, color=Treatment)) +   geom_point(size=3) + geom_smooth(method = lm, aes(group=1))  + theme_bw() + scale_color_manual(values=c("#b2df8a","black", "#a6cee3","#1f78b4")) + theme(legend.position = "none",panel.grid = element_blank())
dev.off()

pdf(file=paste("wgcna_output_files/phenotype_TotalVent_12kPa_MEyellow_",run_date,".pdf",sep=""),h=4, w=3)
ggplot(ME_info, aes(x=MEyellow, y=Total._ventilation_ml_per_g_per_min_12kPa, shape=Treatment, color=Treatment)) +   geom_point(size=3) + geom_smooth(method = lm, aes(group=1))  + theme_bw() + scale_color_manual(values=c("#b2df8a","black", "#a6cee3","#1f78b4")) + theme(legend.position = "none",panel.grid = element_blank())
dev.off()

pdf(file=paste("wgcna_output_files/phenotype_TidalVol_21kPa_MEblue_",run_date,".pdf",sep=""),h=4, w=3)
ggplot(ME_info, aes(x=MEblue, y=Tidal.volume_ul_per_g_21kPa, shape=Treatment, color=Treatment)) +   geom_point(size=3) + geom_smooth(method = lm, aes(group=1))  + theme_bw() + scale_color_manual(values=c("#b2df8a","black", "#a6cee3","#1f78b4")) + theme(legend.position = "none",panel.grid = element_blank())
dev.off()

pdf(file=paste("wgcna_output_files/phenotype_TotalVent_21kPa_MEblue_",run_date,".pdf",sep=""),h=4, w=3)
ggplot(ME_info, aes(x=MEblue, y=Total._ventilation_ml_per_g_per_min_21kPa, shape=Treatment, color=Treatment)) +   geom_point(size=3) + geom_smooth(method = lm, aes(group=1))  + theme_bw() + scale_color_manual(values=c("#b2df8a","black", "#a6cee3","#1f78b4")) + theme(legend.position = "none",panel.grid = element_blank())
dev.off()

pdf(file=paste("wgcna_output_files/phenotype_TotalVent_21kPa_MEturq_",run_date,".pdf",sep=""),h=4, w=3)
ggplot(ME_info, aes(x=MEturquoise, y=Total._ventilation_ml_per_g_per_min_21kPa, shape=Treatment, color=Treatment)) +   geom_point(size=3) + geom_smooth(method = lm, aes(group=1))  + theme_bw() + scale_color_manual(values=c("#b2df8a","black", "#a6cee3","#1f78b4")) + theme(legend.position = "none",panel.grid = element_blank())
dev.off()

#just one with legend
pdf(file=paste("wgcna_output_files/phenotype_Diaphragm_MEyellow_wLegend",run_date,".pdf",sep=""),h=10, w=6)
ggplot(ME_info, aes(x=MEyellow, y=Diaphragm_perc_body_mass, shape=Treatment, color=Treatment)) +   geom_point(size=3) + geom_smooth(method = lm, aes(group=1))  + theme_bw() + scale_color_manual(values=c("#b2df8a","black", "#a6cee3","#1f78b4")) + theme(panel.grid = element_blank())
dev.off()

```


Test for gene enrichment from WGCNA modules that had significant treatment effect in ANOVA (blue/M1, turquoise/M2, yellow/M4, brown/M3).

```{r GO_enrichment_WGCNA_modules}

# output gene info

all_genes <- geneInfo["Gene"]

# Read in the gene annotation
annot = read.csv(file = "../data_files/pman_mus_orthologs.csv")
all_genes_mus <- merge(all_genes, annot, by.x="Gene",by.y="Gene.stable.ID")

# Run GO enrichment using gProfiler package

for (color in c("blue","turquoise","yellow","brown")){
  genes <- geneInfo[geneInfo$moduleColor==color,]["Gene"]
  genes_mus <- merge(genes, annot, by.x="Gene", by.y="Gene.stable.ID")
  genes_GO <- gost(as.vector(genes_mus$Mouse.gene.name), organism = "mmusculus",
                        ordered_query = F, significant = T, exclude_iea = F,
                        user_threshold=0.05,
                        correction_method = "g_SCS", custom_bg = as.vector(all_genes_mus$Mouse.gene.name),
                        domain_scope = "annotated", numeric_ns = "",sources=c("GO","KEGG","REAC","TF","HP"))
  write.csv(apply(genes_GO$result,2,as.character),file=paste("wgcna_output_files/",color,"_module_GO_",run_date,".csv",sep=""))
}


```

Next, we use EdgeR to test for differential gene expression between treatment and control groups using the GLM approach. We will compare the treatment groups, based on page 31 of edgeR manual. We will use the glmQLFit quasi-likelihood tests because they are more conservative and rigorous than the glmFit for measuring differential expression. We identify a "lifelong" contrast of normoxia vs. life-long hypoxia treatments, a "postNatal" contrast of normoxia vs. post-natal hypoxia treatments, and an "adult" contrast of normoxia vs control hypoxia treatments. 

```{r edgeR_analysis_DE}

# check the design,the 0+ in the model formula is an instruction not to 
# include an intercept column and instead to include a column for each group.
#design

fit <- glmQLFit(y, design) # quasi-likelihood F-tests, followed by glmQLFTest()

my.contrasts <- makeContrasts(LifeLong=N-LH, PostNatal=N-PNH, Adult=N-AH, levels=design)

qlf.LifeLong<- glmQLFTest(fit, contrast=my.contrasts[,"LifeLong"])
top1 <- topTags(qlf.LifeLong,p.value=0.05,n=Inf)
top1ids <- top1$table[top1$table$FDR<=0.05 & (top1$table$logFC>=1 | top1$table$logFC<=-1),]

qlf.PostNatal <- glmQLFTest(fit, contrast=my.contrasts[,"PostNatal"])
top2 <- topTags(qlf.PostNatal,p.value=0.05,n=Inf)
top2ids <- top2$table[top2$table$FDR<=0.05 & (top2$table$logFC>=1 | top2$table$logFC<=-1),]

qlf.Adult <- glmQLFTest(fit, contrast=my.contrasts[,"Adult"])
top3 <- topTags(qlf.Adult,p.value=0.05,n=Inf)
top3ids <- top3$table[top3$table$FDR<=0.05 & (top3$table$logFC>=1 | top3$table$logFC<=-1),]


# LIFE-LONG HYPOXIA
summary(de1 <- decideTestsDGE(qlf.LifeLong)) # The total number of DE genes identified at an FDR of 5% can be shown with decideTestsDGE.
detags1 <- rownames(y)[as.logical(de1)]
pdf(paste("edgeR_output_files/smearPlot_lifeLongHypoxia_de_",run_date,".pdf",sep=""),w=7,h=6)
plotSmear(qlf.LifeLong, de.tags=detags1,main="Life-long hypoxia vs. Control")
abline(h=c(-1,1), col='blue')
lifeLong_table = data.frame(topTags(qlf.LifeLong, n=length(keep_counts[,1])))
sig.lifeLong = subset(lifeLong_table, lifeLong_table$PValue <= 0.05) # table of DE genes for pop
sig.lifeLong.fdr <- subset(lifeLong_table, lifeLong_table$FDR <= 0.05) 
dev.off()

plotSmear(qlf.LifeLong, de.tags=detags1,main="Life-long hypoxia vs. Control")
abline(h=c(-1,1), col='blue')

dim(sig.lifeLong.fdr) # sig. effects of lifelong period
sig.lifeLong.fdr_up <- rownames(sig.lifeLong.fdr[sig.lifeLong.fdr$logFC>0,]) # up-regulated genes
sig.lifeLong.fdr_down <- rownames(sig.lifeLong.fdr[sig.lifeLong.fdr$logFC<0,]) # down-regulated genes

# POST-NATAL HYPOXIA
summary(de2 <- decideTestsDGE(qlf.PostNatal)) # The total number of DE genes identified at an FDR of 5% can be shown with decideTestsDGE.
detags2 <- rownames(y)[as.logical(de2)]
pdf(paste("edgeR_output_files/smearPlot_postNatalHypoxia_de_",run_date,".pdf",sep=""),w=7,h=6)
plotSmear(qlf.PostNatal, de.tags=detags2,main="Post-natal Hypoxia vs. Control")
abline(h=c(-1,1), col='blue')
postNatal_table = data.frame(topTags(qlf.PostNatal, n=length(keep_counts[,1])))
sig.postNatal = subset(postNatal_table, postNatal_table$PValue <= 0.05) # table of DE genes for pop
sig.postNatal.fdr <- subset(postNatal_table, postNatal_table$FDR <= 0.05) 
dev.off()

plotSmear(qlf.PostNatal, de.tags=detags2,main="Post-natal Hypoxia vs. Control")
abline(h=c(-1,1), col='blue')

dim(sig.postNatal.fdr) # sig. effects of post-natal hypoxia
sig.postNatal.fdr_up <- rownames(sig.postNatal.fdr[sig.postNatal.fdr$logFC>0,]) # up-regulated genes
sig.postNatal.fdr_down <- rownames(sig.postNatal.fdr[sig.postNatal.fdr$logFC<0,]) # down-regulated genes 

# ADULT HYPOXIA
summary(de3 <- decideTestsDGE(qlf.Adult)) # The total number of DE genes identified at an FDR of 5% can be shown with decideTestsDGE.
detags3 <- rownames(y)[as.logical(de3)]
pdf(paste("edgeR_output_files/smearPlot_adultHypoxia_de_",run_date,".pdf",sep=""),w=7,h=6)
plotSmear(qlf.Adult, de.tags=detags3,main="Adult Hypoxia vs. Control")
abline(h=c(-1,1), col='blue')
adult_table = data.frame(topTags(qlf.Adult, n=length(keep_counts[,1])))
sig.adult = subset(adult_table, adult_table$PValue <= 0.05) # table of DE genes for pop
sig.adult.fdr <- subset(adult_table, adult_table$FDR <= 0.05) 
dev.off()

plotSmear(qlf.Adult, de.tags=detags3,main="Adult Hypoxia vs. Control")
abline(h=c(-1,1), col='blue')

dim(sig.adult.fdr) # sig. effects of post-natal hypoxia
sig.adult.fdr_up <- rownames(sig.adult.fdr[sig.adult.fdr$logFC>0,]) # up-regulated genes
sig.adult.fdr_down <- rownames(sig.adult.fdr[sig.adult.fdr$logFC<0,]) # up-regulated genes

```

```{r edgeR_analysis_intersections}

# What are the genes that intersect between these three comparisons? 

listInput <- list(LifeLongHypoxia = rownames(sig.lifeLong.fdr), PostNatalHypoxia = rownames(sig.postNatal.fdr), AdultHypoxia = rownames(sig.adult.fdr))

pdf(paste("edgeR_output_files/Intersection_of_DE_Genes_",run_date,".pdf",sep=""),w=8,h=6)
upset(fromList(listInput), order.by = "freq", mainbar.y.label="DE Genes", sets.x.label = "DE Genes Per Stage Comparison", sets.bar.color = "#56B4E9")
dev.off()

upset(fromList(listInput), order.by = "freq", mainbar.y.label="DE Genes", sets.x.label = "DE Genes Per Stage Comparison", sets.bar.color = "#56B4E9")



myCol <- c("#CD69C9","#1f78b4","#b2df8a")

# Chart to file
venn.diagram(x = listInput, category.names = c("Lifelong" , "PostNatal " , "Adult"), filename = paste("edgeR_output_files/venn_diagramm_of_DE_genes_",run_date,".png",sep=""), output=TRUE, imagetype="png" , height = 480 , width = 480 , resolution = 300, compression = "lzw", lwd = 2, lty = 'blank', fill = myCol, cex = .6, fontface = "bold", fontfamily = "sans", cat.cex = 0.6, cat.fontface = "bold", cat.default.pos = "outer", cat.pos = c(-27, 27, 135), cat.dist = c(0.055, 0.055, 0.085), cat.fontfamily = "sans", rotation = 1)

# Helper function to display Venn diagram
display_venn <- function(x, ...){
  grid.newpage()
  plt <- venn.diagram(x = listInput, category.names = c("Lifelong" , "PostNatal " , "Adult"),filename=NULL, lwd = 0, lty = 'blank', fill = myCol, cex = .6, fontface = "bold", fontfamily = "sans", cat.cex = 0.6, cat.fontface = "bold", cat.default.pos = "outer", cat.pos = c(-27, 27, 135), cat.dist = c(0.055, 0.055, 0.085), cat.fontfamily = "sans", rotation = 1)
  grid.draw(plt)
}

display_venn(plt)

# Genes that intersect and are up- or down-regulated

# three-way
sig.intersect.fdr <- as.data.frame(intersect(intersect(rownames(sig.postNatal.fdr),rownames(sig.adult.fdr)),rownames(sig.lifeLong.fdr)))
colnames(sig.intersect.fdr) <- "Gene"

sig.lifeLong.fdr$Gene <- rownames(sig.lifeLong.fdr)
sig.adult.fdr$Gene <- rownames(sig.adult.fdr)
sig.postNatal.fdr$Gene <- rownames(sig.postNatal.fdr)

tmp1<-merge(sig.intersect.fdr,sig.lifeLong.fdr[,c("Gene","logFC")],by="Gene")
colnames(tmp1) <- c("Gene","logFC.lifeLong")
tmp2 <- merge(tmp1,sig.adult.fdr[,c("Gene","logFC")],by="Gene")
colnames(tmp2) <- c("Gene","logFC.lifeLong","logFC.adult")
intersect_FC <- merge(tmp2,sig.postNatal.fdr[,c("Gene","logFC")],by="Gene")
colnames(intersect_FC) <- c("Gene","logFC.lifeLong","logFC.adult","logFC.postNatal")

write.table(intersect_FC, file=paste("edgeR_output_files/intersectGenes_FC_",run_date,".txt",sep=""), quote=FALSE, sep="\t", row.names=FALSE, col.names=TRUE)

# make pairwise logFC plots of 207 3-way genes

pdf(file=paste("edgeR_output_files/intersectGenes_FC_lifeLongvPostNatal_",run_date,".pdf",sep=""))
plot(intersect_FC$logFC.lifeLong,intersect_FC$logFC.postNatal, main="LifeLong vs. PostNatal (n=207)"); abline(coef=c(0,1)); abline(v=0,lty=2); abline(h=0,lty=2)
dev.off()

pdf(file=paste("edgeR_output_files/allGenes_FC_lifeLongvPostNatal_",run_date,".pdf",sep=""))
plot(qlf.LifeLong$table$logFC,qlf.PostNatal$table$logFC, main="LifeLong vs. PostNatal (n=15675)"); abline(coef=c(0,1)); abline(v=0,lty=2); abline(h=0,lty=2)
dev.off()

pdf(file=paste("edgeR_output_files/intersectGenes_FC_lifeLongvAdult_",run_date,".pdf",sep=""))
plot(intersect_FC$logFC.lifeLong,intersect_FC$logFC.adult, main="LifeLong vs. Adult (n=207)"); abline(coef=c(0,1)); abline(v=0,lty=2); abline(h=0,lty=2)
dev.off()

# make boxplot of logFC for 207 intersection genes
boxplot(intersect_FC[,2:4],ylab="logFC relative to control")

# make pairwise plots of logFC between all genes

pdf(file=paste("edgeR_output_files/allGenes_FC_lifeLongvAdult_",run_date,".pdf",sep=""))
plot(qlf.LifeLong$table$logFC,qlf.Adult$table$logFC, main="LifeLong vs. Adult (n=15675)"); abline(coef=c(0,1)); abline(v=0,lty=2); abline(h=0,lty=2)
dev.off()

pdf(file=paste("edgeR_output_files/intersectGenes_FC_postNatalvAdult_",run_date,".pdf",sep=""))
plot(intersect_FC$logFC.postNatal,intersect_FC$logFC.adult, main="PostNatal vs. Adult (n=207)"); abline(coef=c(0,1)); abline(v=0,lty=2); abline(h=0,lty=2)
dev.off()

pdf(file=paste("edgeR_output_files/allGenes_FC_postNatalvAdult_",run_date,".pdf",sep=""))
plot(qlf.PostNatal$table$logFC,qlf.Adult$table$logFC, main="PostNatal vs. Adult (n=15675)"); abline(coef=c(0,1)); abline(v=0,lty=2); abline(h=0,lty=2)
dev.off()

# make pairwise plots of logFC of unions between sets

dim(lifeLong_table); dim(postNatal_table); dim(adult_table) # can merge by gene
lifeLong_table$Gene=rownames(lifeLong_table)
postNatal_table$Gene=rownames(postNatal_table)
adult_table$Gene=rownames(adult_table)

tmp_table <- merge(lifeLong_table, postNatal_table, by="Gene",suffixes = c(".LH",".PNH"))
allThree_table <- merge(tmp_table,adult_table,by="Gene",suffixes=c("",".AH"))

allThree_table <- allThree_table[,c(1,2,6,7,11,12,16)]
colnames(allThree_table)[6:7] <- c("logFC.AH","FDR.AH")

# MAKE BIG PLOT OF 3 PAIRWISE LOGFC PLOTS
pdf(file=paste("edgeR_output_files/Figure_X_logFC_pairwiseAll_",run_date,".pdf",sep=""),height=6,width=12)
par(mfrow=c(1,3))
# lH, PNH, AH myCol <- c("#440154FF","#1f78b4","#b2df8a")
# for LH v AH
plot(allThree_table$logFC.LH,allThree_table$logFC.AH,pch=".",col="gray",xlab="logFC Lifelong Hypoxia",ylab="logFC Adult Hypoxia",xlim=c(-4,6),ylim=c(-4,6),cex.lab=2)
abline(coef=c(0,1)); abline(v=0,lty=2); abline(h=0,lty=2)
for (i in c(1:dim(allThree_table)[1])){
  tmp_row <- allThree_table[i,]
  if (tmp_row$FDR.LH<=0.05 & tmp_row$FDR.AH<=0.05){
    points(tmp_row$logFC.LH,tmp_row$logFC.AH,pch=16,col="darkgreen",cex=0.8) # intersect DE
  }
  if (tmp_row$FDR.LH<=0.05 & tmp_row$FDR.AH>0.05){
    points(tmp_row$logFC.LH,tmp_row$logFC.AH,pch=17,col=myCol[1],cex=0.8) # just LH DE
  }
  if (tmp_row$FDR.LH>0.05 & tmp_row$FDR.AH<=0.05){
    points(tmp_row$logFC.LH,tmp_row$logFC.AH,pch=18,col=myCol[3],cex=0.8) # just AH DE
  }
}
legend("bottomright",legend=c("Intersection", "Lifelong DE","Adult DE"),
       col=c("darkgreen", myCol[1],myCol[3]), pch=c(16,17,18,cex=0.8))

# for PNH v AH
plot(allThree_table$logFC.PNH,allThree_table$logFC.AH,pch=".",col="gray",xlab="logFC Post-natal Hypoxia",ylab="logFC Adult Hypoxia",xlim=c(-4,6),ylim=c(-4,6),cex.lab=2)
abline(coef=c(0,1)); abline(v=0,lty=2); abline(h=0,lty=2)
for (i in c(1:dim(allThree_table)[1])){
  tmp_row <- allThree_table[i,]
  if (tmp_row$FDR.PNH<=0.05 & tmp_row$FDR.AH<=0.05){
    points(tmp_row$logFC.PNH,tmp_row$logFC.AH,pch=16,col="darkgreen",cex=0.8) # intersect DE
  }
  if (tmp_row$FDR.PNH<=0.05 & tmp_row$FDR.AH>0.05){
    points(tmp_row$logFC.PNH,tmp_row$logFC.AH,pch=17,col=myCol[2],cex=0.8) # just PNH DE
  }
  if (tmp_row$FDR.PNH>0.05 & tmp_row$FDR.AH<=0.05){
    points(tmp_row$logFC.PNH,tmp_row$logFC.AH,pch=18,col=myCol[3],cex=0.8) # just AH DE
  }
}
legend("bottomright",legend=c("Intersection", "Post-natal DE","Adult DE"),
       col=c("darkgreen", myCol[2],myCol[3]), pch=c(16,17,18,cex=0.8))

# for LH v PNH
plot(allThree_table$logFC.LH,allThree_table$logFC.PNH,pch=".",col="gray",xlab="logFC Lifelong Hypoxia",ylab="logFC Post-natal Hypoxia",xlim=c(-4,6),ylim=c(-4,6),cex.lab=2)
abline(coef=c(0,1)); abline(v=0,lty=2); abline(h=0,lty=2)
for (i in c(1:dim(allThree_table)[1])){
  tmp_row <- allThree_table[i,]
  if (tmp_row$FDR.LH<=0.05 & tmp_row$FDR.PNH<=0.05){
    points(tmp_row$logFC.LH,tmp_row$logFC.PNH,pch=16,col="darkgreen",cex=0.8) # intersect DE
  }
  if (tmp_row$FDR.LH<=0.05 & tmp_row$FDR.PNH>0.05){
    points(tmp_row$logFC.LH,tmp_row$logFC.PNH,pch=17,col=myCol[1],cex=0.8) # just LH DE
  }
  if (tmp_row$FDR.LH>0.05 & tmp_row$FDR.PNH<=0.05){
    points(tmp_row$logFC.LH,tmp_row$logFC.PNH,pch=18,col=myCol[2],cex=0.8) # just PNH DE
  }
}
legend("bottomright",legend=c("Intersection", "Lifelong DE","Post-natal DE"),
       col=c("darkgreen", myCol[1],myCol[2]), pch=c(16,17,18,cex=0.8))
dev.off()



union_lifelong_adult_genes <- merge(sig.adult.fdr,sig.lifeLong.fdr[,c("Gene","logFC")],by="Gene",all.x=TRUE,all.y=TRUE)["Gene"]
qlf.LifeLong_subset <- qlf.LifeLong$table
qlf.LifeLong_subset$Gene=rownames(qlf.LifeLong_subset)

qlf.Adult_subset <- qlf.Adult$table
qlf.Adult_subset$Gene=rownames(qlf.Adult_subset)

union_lifelong_adult_genes_logFC <- merge(union_lifelong_adult_genes,qlf.Adult_subset,by="Gene")
union_lifelong_adult_genes_logFC_final <- merge(union_lifelong_adult_genes_logFC,qlf.LifeLong_subset,by="Gene",suffixes = c(".adult",".lifelong"))



plot(union_lifelong_adult_genes_logFC_final$logFC.lifelong, union_lifelong_adult_genes_logFC_final$logFC.adult, main="LifeLong vs. Adult (n=6415)"); abline(coef=c(0,1)); abline(v=0,lty=2); abline(h=0,lty=2)


sig.lifeLong.fdr$Gene <- rownames(sig.lifeLong.fdr)
sig.adult.fdr$Gene <- rownames(sig.adult.fdr)
sig.postNatal.fdr$Gene <- rownames(sig.postNatal.fdr)


plot(qlf.LifeLong$table$logFC,qlf.Adult$table$logFC, main="LifeLong vs. Adult (n=15675)"); abline(coef=c(0,1)); abline(v=0,lty=2); abline(h=0,lty=2)


# Exclusive LH genes (n=1)

sig.intersect_LH_AH <- intersect(sig.lifeLong.fdr$Gene,sig.adult.fdr$Gene)
sig.intersect_LH_PNH <- intersect(sig.lifeLong.fdr$Gene,sig.postNatal.fdr$Gene)
sig.intersect_AH_PNH <- intersect(sig.adult.fdr$Gene,sig.postNatal.fdr$Gene)

only_LH <- setdiff(sig.lifeLong.fdr$Gene,union(sig.adult.fdr$Gene,sig.postNatal.fdr$Gene)) #ENSPEMG00000027446

only_PNH <- setdiff(sig.postNatal.fdr$Gene,union(sig.adult.fdr$Gene,sig.lifeLong.fdr$Gene))

only_dev <- setdiff(union(sig.lifeLong.fdr$Gene,sig.postNatal.fdr$Gene),union(sig.intersect_LH_AH,sig.intersect_AH_PNH))

```



And, finally, we run GO enrichment analysis on the sets of genes from edgeR.

```{r edgeR_analysis_GO_enrichment}

#background will be all genes with keep_counts 
all_genes <- as.data.frame(row.names(keep_counts))
colnames(all_genes) <- c("Gene")
all_genes_mus <- merge(all_genes, annot, by.x="Gene",by.y="Gene.stable.ID")

#lifelong GO
sig.lifeLong.fdr$Gene <- rownames(sig.lifeLong.fdr)
sig.lifeLong.fdr.mus <- merge(sig.lifeLong.fdr, annot, by.x="Gene",by.y="Gene.stable.ID")
write.csv(sig.lifeLong.fdr.mus, file=paste("edgeR_output_files/lifeLongHypoxia_de_genes_fdr_",run_date,".csv",paste=""), quote=FALSE)

lifeLong_genes_GO <- gost(as.vector(sig.lifeLong.fdr.mus $Mouse.gene.stable.ID), organism = "mmusculus",
                           ordered_query = F, significant = T, exclude_iea = F,
                           user_threshold=0.05,
                           correction_method = "g_SCS", custom_bg = as.vector(all_genes_mus$Mouse.gene.stable.ID),
                           domain_scope = "annotated", numeric_ns = "",sources=c("GO","KEGG","REAC","TF","HP"))
#write GO table
write.csv(apply(lifeLong_genes_GO$result,2,as.character),file=paste("edgeR_output_files/lifeLongHypoxia_genes_GO_",run_date,".csv",sep=""))

# post natal dev GO
sig.postNatal.fdr$Gene <- rownames(sig.postNatal.fdr)
sig.postNatal.fdr.mus <- merge(sig.postNatal.fdr, annot, by.x="Gene",by.y="Gene.stable.ID")
write.csv(sig.postNatal.fdr.mus, file=paste("edgeR_output_files/postNatalHypoxia_de_genes_",run_date,".csv",sep=""), quote=FALSE)

postNatal_genes_GO <- gost(as.vector(sig.postNatal.fdr.mus $Mouse.gene.stable.ID), organism = "mmusculus",
                              ordered_query = F, significant = T, exclude_iea = F,
                              user_threshold=0.05,
                              correction_method = "g_SCS", custom_bg = as.vector(all_genes_mus$Mouse.gene.stable.ID),
                              domain_scope = "annotated", numeric_ns = "",sources=c("GO","KEGG","REAC","TF","HP"))
#write go table
write.csv(apply(postNatal_genes_GO$result,2,as.character),file=paste("edgeR_output_files/postNatalHypoxia_genes_GO_",run_date,".csv",sep=""))

# intersection between three comparisons

sig.intersect.fdr.mus <- merge(sig.intersect.fdr, annot, by.x="Gene",by.y="Gene.stable.ID")
write.csv(sig.intersect.fdr.mus, file=paste("edgeR_output_files/intersect_de_genes_",run_date,".csv",sep=""), quote=FALSE)

intersect_genes_GO <- gost(as.vector(sig.intersect.fdr.mus $Mouse.gene.stable.ID), organism = "mmusculus",
                           ordered_query = F, significant = T, exclude_iea = F,
                           user_threshold=0.05,
                           correction_method = "g_SCS", custom_bg = as.vector(all_genes_mus$Mouse.gene.stable.ID),
                           domain_scope = "annotated", numeric_ns = "",sources=c("GO","KEGG","REAC","TF","HP"))

#write go table
write.csv(apply(intersect_genes_GO$result,2,as.character),file=paste("edgeR_output_files/intersect_genes_GO_",run_date,".csv",sep=""))


# only PNH

only_PNH.mus <- merge(as.data.frame(only_PNH), annot, by.x="only_PNH",by.y="Gene.stable.ID")
write.csv(only_PNH.mus, file=paste("edgeR_output_files/only_PNH_de_genes_fdr_",run_date,".csv",paste=""), quote=FALSE)

only_PNH_genes_GO <- gost(as.vector(only_PNH.mus $Mouse.gene.stable.ID), organism = "mmusculus",
                           ordered_query = F, significant = T, exclude_iea = F,
                           user_threshold=0.05,
                           correction_method = "g_SCS", custom_bg = as.vector(all_genes_mus$Mouse.gene.stable.ID),
                           domain_scope = "annotated", numeric_ns = "",sources=c("GO","KEGG","REAC","TF","HP"))

#write go table
write.csv(apply(only_PNH_genes_GO$result,2,as.character),file=paste("edgeR_output_files/only_PNH_genes_GO_",run_date,".csv",sep=""))

# only developmental

only_dev.mus <- merge(as.data.frame(only_dev), annot, by.x="only_dev",by.y="Gene.stable.ID")
write.csv(only_PNH.mus, file=paste("edgeR_output_files/only_PNH_de_genes_fdr_",run_date,".csv",paste=""), quote=FALSE)

only_dev_genes_GO <- gost(as.vector(only_dev.mus $Mouse.gene.stable.ID), organism = "mmusculus",
                           ordered_query = F, significant = T, exclude_iea = F,
                           user_threshold=0.05,
                           correction_method = "g_SCS", custom_bg = as.vector(all_genes_mus$Mouse.gene.stable.ID),
                           domain_scope = "annotated", numeric_ns = "",sources=c("GO","KEGG","REAC","TF","HP"))

#write go table
write.csv(apply(only_dev_genes_GO$result,2,as.character),file=paste("edgeR_output_files/only_dev_genes_GO_",run_date,".csv",sep=""))

# only adult hypoxia 

sig.adult.fdr$Gene <- rownames(sig.adult.fdr)
sig.adult.fdr.mus <- merge(sig.adult.fdr, annot, by.x="Gene",by.y="Gene.stable.ID")
write.csv(sig.adult.fdr.mus, file=paste("edgeR_output_files/adultHypoxia_de_genes_",run_date,".csv",sep=""), quote=FALSE)

adult_genes_GO <- gost(as.vector(sig.adult.fdr.mus $Mouse.gene.stable.ID), organism = "mmusculus",
                              ordered_query = F, significant = T, exclude_iea = F,
                              user_threshold=0.05,
                              correction_method = "g_SCS", custom_bg = as.vector(all_genes_mus$Mouse.gene.stable.ID),
                              domain_scope = "annotated", numeric_ns = "",sources=c("GO","KEGG","REAC","TF","HP"))
#write go table
write.csv(apply(adult_genes_GO$result,2,as.character),file=paste("edgeR_output_files/adultHypoxia_genes_GO_",run_date,".csv",sep=""))




```

We are also interested in which genes that are differentially-expressed from edgeR analysis intersect with modules from WGCNA. WGCNA modules are non-overlapping, so we will examine overlap of M1, M2, M3, M4 each with the DE gene sets. 

```{r intersect DE genes with WGCNA}

# intersect w/M1(blue)  
  genes <- geneInfo[geneInfo$moduleColor=="blue",]["Gene"]
  listInput <- list(LifeLongHypoxia = rownames(sig.lifeLong.fdr), PostNatalHypoxia = rownames(sig.postNatal.fdr), AdultHypoxia = rownames(sig.adult.fdr),WGCNA_M1 = genes$Gene)
  pdf(paste("edgeR_output_files/Intersection_of_DE_Genes_wWGCNA_M1_",run_date,".pdf",sep=""),w=8,h=6)
  upset(fromList(listInput), order.by = "freq", mainbar.y.label="Gene Intersections", sets.x.label = "DE Genes Comparison w/WGCNA Module", sets.bar.color = "#56B4E9")
  dev.off()

# intersect w/M2 (turquoise)
  genes <- geneInfo[geneInfo$moduleColor=="turquoise",]["Gene"]
  listInput <- list(LifeLongHypoxia = rownames(sig.lifeLong.fdr), PostNatalHypoxia = rownames(sig.postNatal.fdr), AdultHypoxia = rownames(sig.adult.fdr),WGCNA_M2 = genes$Gene)
  pdf(paste("edgeR_output_files/Intersection_of_DE_Genes_wWGCNA_M2_",run_date,".pdf",sep=""),w=8,h=6)
  upset(fromList(listInput), order.by = "freq", mainbar.y.label="Gene Intersections", sets.x.label = "DE Genes Comparison w/WGCNA Module", sets.bar.color = "#56B4E9")
  dev.off()
  
# intersect w/M3 (brown)
  genes <- geneInfo[geneInfo$moduleColor=="brown",]["Gene"]
  listInput <- list(LifeLongHypoxia = rownames(sig.lifeLong.fdr), PostNatalHypoxia = rownames(sig.postNatal.fdr), AdultHypoxia = rownames(sig.adult.fdr),WGCNA_M3 = genes$Gene)
  pdf(paste("edgeR_output_files/Intersection_of_DE_Genes_wWGCNA_M3_",run_date,".pdf",sep=""),w=8,h=6)
  upset(fromList(listInput), order.by = "freq", mainbar.y.label="Gene Intersections", sets.x.label = "DE Genes Comparison w/WGCNA Module", sets.bar.color = "#56B4E9")
  dev.off()

# intersect w/M4 (yellow)
  genes <- geneInfo[geneInfo$moduleColor=="yellow",]["Gene"]
  listInput <- list(LifeLongHypoxia = rownames(sig.lifeLong.fdr), PostNatalHypoxia = rownames(sig.postNatal.fdr), AdultHypoxia = rownames(sig.adult.fdr),WGCNA_M4 = genes$Gene)
  pdf(paste("edgeR_output_files/Intersection_of_DE_Genes_wWGCNA_M4_",run_date,".pdf",sep=""),w=8,h=6)
  upset(fromList(listInput), order.by = "freq", mainbar.y.label="Gene Intersections", sets.x.label = "DE Genes Comparison w/WGCNA Module", sets.bar.color = "#56B4E9")
  dev.off()
  
    
```

Our final aim in this study is to identify which differentially-expressed genes/gene sets (from WGCNA and edgeR) overlap w/candidate selection scan genes, and whether the overlap is a significant enrichment.  

```{r overlap_w_selection_scan}

# Open file with selection outliers

sel_genes <- read.table("../data_files/PBS_999_genes_pman_mus_convert.txt",header=T) # 1852 genes

# merge all expressed genes with annotation file

keep_counts_genes <- as.data.frame(rownames(keep_counts))
colnames(keep_counts_genes) <- c("gene_pman_ensembl")

keep_counts_genes_mus <- merge(keep_counts_genes,annot,by.x="gene_pman_ensembl",by.y="Gene.stable.ID",all.x=TRUE)

sel_genes_mus <- merge(sel_genes,keep_counts_genes_mus,by.x="gene_mus",by.y="Mouse.gene.name",all.x=TRUE) #1538 genes

# How many genes are in blue, turquoise, yellow, brown modules and also in PBS? 
for (color in c("blue","turquoise","yellow","brown")){
  genes <- geneInfo[geneInfo$moduleColor==color,]["Gene"]
  print(color)
  print(length(intersect(genes$Gene, sel_genes_mus$gene_pman_ensembl)))
  #print(head(intersect(genes$Gene, sel_genes_mus$gene_pman_ensembl)))
  tmp_mus <- merge(genes,sel_genes_mus, by.x="Gene", by.y="gene_pman_ensembl")
  tmp_mus_GO <- gost(as.vector(tmp_mus $Mouse.gene.stable.ID), organism = "mmusculus",
                              ordered_query = F, significant = T, exclude_iea = F,
                              user_threshold=0.05,
                              correction_method = "g_SCS", custom_bg = as.vector(all_genes_mus$Mouse.gene.stable.ID),
                              domain_scope = "annotated", numeric_ns = "",sources=c("GO","KEGG","REAC","TF","HP"))
  #write go table
  write.csv(apply(tmp_mus_GO$result,2,as.character),file=paste("selection_overlap_output_files/WGCNA_",color,"_selOverlap_GO",run_date,".csv",sep=""))
  write.csv(tmp_mus[!duplicated(tmp_mus[c('Gene')]), ],file=paste("selection_overlap_output_files/WGCNA_",color,"_selOverlap_genes",run_date,".csv",sep=""))
}

# Are those sig. enrichments? 

dat_M1 <- data.frame(
     "selected_yes" = c(662, 1190),
     "selected_no" = c(5211, 8612),
     row.names = c("M1", "Non-M1"),
     stringsAsFactors = FALSE
 )
colnames(dat_M1) <- c("Selected", "Not-selected")
dat_M1
fisher.test(dat_M1)

dat_M2 <- data.frame(
     "selected_yes" = c(407, 1445),
     "selected_no" = c(5626, 8197),
     row.names = c("M2", "Non-M2"),
     stringsAsFactors = FALSE
 )
colnames(dat_M2) <- c("Selected", "Not-selected")
dat_M2
fisher.test(dat_M2)

dat_M4 <- data.frame(
     "selected_yes" = c(63, 1789),
     "selected_no" = c(498, 13327),
     row.names = c("M4", "Non-M4"),
     stringsAsFactors = FALSE
 )
colnames(dat_M4) <- c("Selected", "Not-selected")
dat_M4
fisher.test(dat_M4)

dat_M3 <- data.frame(
     "selected_yes" = c(150, 1702),
     "selected_no" = c(1105, 12718),
     row.names = c("M3", "Non-M3"),
     stringsAsFactors = FALSE
 )
colnames(dat_M3) <- c("Selected", "Not-selected")
dat_M3
fisher.test(dat_M3)

chooseTopHubInEachModule(Expr, moduleColors,power=13)

# How about genes that overlap with set of 209 DE dev genes? 

print(length(intersect(only_dev.mus$only_dev, sel_genes_mus$gene_pman_ensembl))) #16
selected_dev_genes <- merge(only_dev.mus, sel_genes_mus,by.x="only_dev",by.y="gene_pman_ensembl")
write.csv(selected_dev_genes,file=paste("selection_overlap_output_files/only_dev_genes_selCandidates_",run_date,".csv",sep=""))

## Is that an enrichment? 

dat_dev <- data.frame(
     "selected_yes" = c(16, 1836),
     "selected_no" = c(193, 13630),
     row.names = c("dev DE", "Non-dev DE"),
     stringsAsFactors = FALSE
 )
colnames(dat_dev) <- c("Selected", "Not-selected")
dat_dev
fisher.test(dat_dev)


# get expression CPM values 
#sel_dev_genes_cpm <- as.data.frame(t(diaphragm.norm[unique(selected_dev_genes$only_dev),]))
#treatments <- c(rep("4.AH",8),rep("2.LH",8),rep("1.C",10),rep("3.PNH",9))
#sel_dev_genes_cpm$treatment <- treatments

#ggplot(sel_dev_genes_cpm, aes(x=treatment, y=ENSPEMG00000025708)) + ggtitle("ENSPEMG00000025708") + 
#  geom_boxplot() + geom_dotplot(binaxis='y', stackdir='center',
#  position=position_dodge(1), dotsize = 0.2)+ ylab("Relative gene expression (log 2 counts per million))")


# How many genes overlap with the 3-way intersection set of 207?
print(length(intersect(sig.intersect.fdr.mus$Gene, sel_genes_mus$gene_pman_ensembl))) #23
selected_threeWay_genes <- merge(sig.intersect.fdr.mus, sel_genes_mus,by.x="Gene",by.y="gene_pman_ensembl")
write.csv(selected_threeWay_genes,file=paste("selection_overlap_output_files/threeWay_genes_selCandidates_",run_date,".csv",sep=""))

## Is that an enrichment? 

dat_3way <- data.frame(
     "selected_yes" = c(23, 1829),
     "selected_no" = c(184, 13639),
     row.names = c("3way DE", "Non-3way DE"),
     stringsAsFactors = FALSE
 )
colnames(dat_3way) <- c("Selected", "Not-selected")
dat_3way
fisher.test(dat_3way)

# How many genes are in EdgeR DE gene sets and also in PBS? Is that an enrichment? 
# 1852 pman genes expressed and in PBS 99.9th

# LH DE genes 
dim(sig.lifeLong.fdr)[1] #value of significant fdr genes
print(length(intersect(sig.lifeLong.fdr$Gene, sel_genes_mus$gene_pman_ensembl)))

dat_LH <- data.frame(
     "selected_yes" = c(27, 1825),
     "selected_no" = c(195, 13628),
     row.names = c("LH DE", "Non-LH DE"),
     stringsAsFactors = FALSE
 )
colnames(dat_LH) <- c("Selected", "Not-selected")
dat_LH
fisher.test(dat_LH)

# not enriched so no GO but output list of genes
tmp_mus <- merge(sig.lifeLong.fdr,sel_genes_mus, by.x="Gene", by.y="gene_pman_ensembl")
write.csv(tmp_mus[!duplicated(tmp_mus[c('Gene')]), ],file=paste("selection_overlap_output_files/DE_lifelongHypoxia_selOverlap_genes",run_date,".csv",sep=""))

# AH DE genes 
dim(sig.adult.fdr)[1] #value of significant fdr genes
print(length(intersect(sig.adult.fdr$Gene, sel_genes_mus$gene_pman_ensembl)))

dat_AH <- data.frame(
     "selected_yes" = c(660, 5749),
     "selected_no" = c(1192, 8074),
     row.names = c("AH DE", "Non-AH DE"),
     stringsAsFactors = FALSE
 )
colnames(dat_AH) <- c("Selected", "Not-selected")
dat_AH
fisher.test(dat_AH)

# sig enrichment, so run GO analysis
tmp_mus <- merge(sig.adult.fdr,sel_genes_mus, by.x="Gene", by.y="gene_pman_ensembl")
tmp_mus_GO <- gost(as.vector(tmp_mus $Mouse.gene.stable.ID), organism = "mmusculus",
                              ordered_query = F, significant = T, exclude_iea = F,
                              user_threshold=0.05,
                              correction_method = "g_SCS", custom_bg = as.vector(all_genes_mus$Mouse.gene.stable.ID),
                              domain_scope = "annotated", numeric_ns = "",sources=c("GO","KEGG","REAC","TF","HP"))
  #write go table
  write.csv(apply(tmp_mus_GO$result,2,as.character),file=paste("selection_overlap_output_files/DE_adultHypoxia_selOverlap_GO",run_date,".csv",sep=""))
  write.csv(tmp_mus[!duplicated(tmp_mus[c('Gene')]), ],file=paste("selection_overlap_output_files/DE_adultHypoxia_selOverlap_genes",run_date,".csv",sep=""))

# PNH DE genes 
dim(sig.postNatal.fdr)[1] #value of significant fdr genes
print(length(intersect(sig.postNatal.fdr$Gene, sel_genes_mus$gene_pman_ensembl)))

dat_PNH <- data.frame(
     "selected_yes" = c(275, 2644),
     "selected_no" = c(1577, 11179),
     row.names = c("PNH DE", "Non-PNH DE"),
     stringsAsFactors = FALSE
 )
colnames(dat_PNH) <- c("Selected", "Not-selected")
dat_PNH
fisher.test(dat_PNH)

# sig enrichment so run GO analysis

tmp_mus <- merge(sig.postNatal.fdr,sel_genes_mus, by.x="Gene", by.y="gene_pman_ensembl")
tmp_mus_GO <- gost(as.vector(tmp_mus $Mouse.gene.stable.ID), organism = "mmusculus",
                              ordered_query = F, significant = T, exclude_iea = F,
                              user_threshold=0.05,
                              correction_method = "g_SCS", custom_bg = as.vector(all_genes_mus$Mouse.gene.stable.ID),
                              domain_scope = "annotated", numeric_ns = "",sources=c("GO","KEGG","REAC","TF","HP"))
  #write go table
  write.csv(apply(tmp_mus_GO$result,2,as.character),file=paste("selection_overlap_output_files/DE_postNatalHypoxia_selOverlap_GO",run_date,".csv",sep=""))
  write.csv(tmp_mus[!duplicated(tmp_mus[c('Gene')]), ],file=paste("selection_overlap_output_files/DE_postNatalHypoxia_selOverlap_genes",run_date,".csv",sep=""))


# What is the proportion of the identified candidates for high-altitude adaptation represented in the DEGs and genes-modules?
  
sel_modules_tmp <- as.data.frame(unique(geneInfo[(geneInfo$moduleColor!="grey" & geneInfo$moduleColor!="green"),1]))
colnames(sel_modules_tmp) <- c("Gene")

length(unique(intersect(sel_genes_mus$gene_pman_ensembl,sel_modules_tmp$Gene)))  # 1282 out of 1435



```


sessionInfo()










